src/machine-learning/machine-learning-api-client.ts

```diff
@@ -140,6 +152,31 @@ export class MachineLearningApiClient {
       });
   }
 
+  public listModels(options: ListModelsOptions = {}): Promise<ListModelsResponse> {
+    if (typeof options.filter !== 'undefined' && !validator.isNonEmptyString(options.filter)) {
+      const err = new FirebaseMachineLearningError('invalid-argument', 'Invalid list filter.');
+      return Promise.reject(err);
+    }
+    if (typeof options.pageSize !== 'undefined' && !validator.isNumber(options.pageSize)) {
```

Comment:
Put a cap on the pageSize.

Reply:
There's a cap on the number of models people are allowed to create, so it automatically functions as a pageSize cap too. Is that sufficient?

Reply:
We usually put an explicit cap on all list operations. This is documented and the inputs are checked and rejected in the SDK if they exceed the cap. I think we need something similar here. You can use the same underlying cap as the backend API.